tools/tsp-client/src/index.ts

```diff
@@ -229,9 +229,17 @@ async function main() {
     rootUrl = resolvePath(options.outputDir);
   }
 
+  // FIXME: this is a workaround meanwhile we fix the issue with failing to delete the sparse-spec directory
+  // Tracking issue: https://github.com/Azure/azure-sdk-tools/issues/7636
+  const repoRoot = await getRepoRoot(rootUrl);
+  access(joinPaths(repoRoot, "..", "sparse-spec")).then(() => {
```

Comment:
Since this is already inside an async function, I would recommend using await and try/catch to keep it consistent and easier to read. Also, do we want to silently swallow the error with an empty catch?

Reply:
consider logging information that can help tsp-config customers send actionable issue reports (maybe there is nothing to log and that's fine, just wanted to mention this)

Reply:
sounds good, will switch to a try/catch. Also, since this is only meant to check if the sparse-spec/ directory exists if there are any other errors related to this, then I didnt want to throw on that. Also, I noticed we get an error if the directory doesnt exist which in this case is what we want, so we're ignoring the error.